core/src/main/java/com/rudderstack/android/sdk/core/RudderPreferenceManager.java

```diff
@@ -107,5 +131,14 @@ String getPeriodicWorkRequestId() {
         return preferences.getString(RUDDER_PERIODIC_WORK_REQUEST_ID_KEY, null);
     }
 
+    void performMigration() {
+        int versionCode = getBuildVersionCode();
+        if (versionCode != -1) {
```

Comment:
Actually we are returning `-1` as a default value when we couldn't when we couldn't find any value in the shared preferences. So this would work even if the version is `0` right ?